rustls/src/client/mod.rs

```diff
@@ -620,6 +617,55 @@ impl ClientSessionImpl {
     }
 }
 
+// Added in anticipation of another struct, EncryptedHost, that returns
+// Some() for the last two methods.
+pub trait HelloData {
```

Comment:
This should help reduce the trait bounds:

```suggestion
pub trait HelloData: Send + Sync + 'static {
```

Reply:
I swear I tried that... but I'll check again. Not in the patch anymore, though.